### PR TITLE
app-containers/syft: add v0.51.0 to sdk

### PR DESCRIPTION
**Pull request by jepio**
# app-containers/syft: add v0.51.0 to sdk

Add syft to our sdk. Syft can be used to create an SBOM from gentoo's package data at `$ROOT/var/db/pkg` (PR to scripts incoming).

## How to use

```
sudo emerge syft
sudo syft /build/amd64-usr -o spdx-json=sbom.json
```

## Testing done

[Describe the testing you have done before submitting this PR. Please include both the commands you issued as well as the output you got.]

- [ ] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)

### CODE REVIEW

**Review by pothos — COMMENTED** [review]
For the future updaters and PR reviewers it would be easier to first import the upstream file in one commit and then do the changes in a second

**Review by JAORMX — APPROVED** [review]

**Comment by jepio** [comment]
> For the future updaters and PR reviewers it would be easier to first import the upstream file in one commit and then do the changes in a second

OK, i'll rework it

File: app-containers/syft/syft-0.51.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+SRC_URI="https://github.com/anchore/syft/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI+=" https://dev.gentoo.org/~williamh/dist/${P}-deps.tar.xz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+# Flatcar: keyword for arm64 and amd64
+KEYWORDS="amd64 arm64"
+
+src_compile() {
+	# Flatcar: add ldflags to set version
```
Comment (tormath1) [inline app-containers/syft/syft-0.51.0.ebuild:18]: Thanks! I think this change can be upstreamed with the `dobin bin/*` indentation fix. We should also submit a keyword request (`~arm64` or `arm64`)

**Review by pothos — APPROVED** [review]

File: app-containers/syft/syft-0.51.0.ebuild (hunk 1 above)
Comment (jepio) [reply app-containers/syft/syft-0.51.0.ebuild:18]: I've opened a PR here: https://github.com/gentoo/gentoo/pull/26617

**Comment by jepio** [comment]
@tormath1 i'll merge this now, and will come back to this once gentoo PR is merged

**Pull request merged**